discretisedfield/mesh.py

```diff
@@ -781,7 +781,7 @@ def point2index(self, point, /):
         if point not in self.region:
             raise ValueError(f"Point {point} is outside the region {self.region=}.")
 
-        index = ((point - self.region.pmin) / self.cell - 0.5).round().astype(int)
+        index = np.floor((point - self.region.pmin) / self.cell).astype(int)
```

Comment:
In the documentation we should mention that cells (apart from the last one) are half-open intervals.